### ISSUE TITLE
Support C2D on confidential networks

### ISSUE THREAD

**Issue by paulo-ocean** [oceanprotocol/ocean-node#692]
Right now this is not fully supported because at least for the "initialize" and "startCompute" operations we need to decrypt the DDO files object.
When we're dealing with a confidential EVM and template 4 we need to get the files object from a smart contract call, and for that we need a consumer signature, which ATM some of these operations are missing

**Comment by mihaisc** [comment]
is this done?

**Comment by paulo-ocean** [comment]
> is this done?

Nope, its not. We will need to refactor some C2D APIs, node side at least.

**Comment by paulo-ocean** [comment]
closing this one (will be part of [https://github.com/oceanprotocol/ocean-node/pull/705](https://github.com/oceanprotocol/ocean-node/pull/705)
